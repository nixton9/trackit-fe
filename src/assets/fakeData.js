export const notes = [
  {
    id: 1,
    title: 'Lista de compras',
    date: '2020-07-20',
    content: 'asdasdad',
    tags: [
      { id: 1, name: 'Primary', color: '#F44336' },
      { id: 2, name: 'Secondary', color: '#E91E63' }
    ]
  },
  {
    id: 2,
    title: 'Apontamentos aula',
    date: '2020-06-25',
    content: 'asdasdad',
    tags: [
      { id: 3, name: 'Studies', color: '#607D8B' },
      { id: 4, name: 'School', color: '#00BCD4' }
    ]
  },
  {
    id: 3,
    title: 'Random shit',
    date: '2020-07-15',
    content: 'asdasdad',
    tags: [{ id: 5, name: 'Random', color: '#3F51B5' }]
  },
  {
    id: 4,
    title: 'Work shit',
    date: '2020-07-15',
    content: 'asdasdad',
    tags: [
      { id: 6, name: 'Work', color: '#9C27B0' },
      { id: 7, name: 'Coding', color: '#FF9800' },
      { id: 8, name: 'Stuff', color: '#4CAF50' }
    ]
  },
  {
    id: 5,
    title: 'Post 129',
    date: '2020-05-07',
    content: 'asdasdad'
  }
]
