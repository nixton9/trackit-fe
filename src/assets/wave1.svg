<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="744" height="744" viewBox="0 0 744 744">
  <defs>
    <clipPath id="clip-Prancheta_7">
      <rect width="744" height="744"/>
    </clipPath>
  </defs>
  <g id="Prancheta_7" data-name="Prancheta â€“ 7" clip-path="url(#clip-Prancheta_7)">
    <path id="Caminho_294" data-name="Caminho 294" d="M-14697.143-11591.146s204.548,93.02,241.44,226.007,101.77,246.085,225.016,271.63,239.81-133.94,286.815-109.888,72.048,136.969,72.048,136.969,18.206,112.27,0,146.063-154.687,113.568-154.687,113.568-145.483,41.1-252.11,41.649-282.684-4.542-332.2-22.729-57.569-16.607-86.319-46.7-74.028-81.864-81.211-123.045-11.661-111.631-8.345-135.894,30.468-240.38,30.94-257.871-.572-130.1,0-136.284S-14697.143-11591.146-14697.143-11591.146Z" transform="translate(14695 11649)" fill="#273150" opacity="0.051"/>
  </g>
</svg>
