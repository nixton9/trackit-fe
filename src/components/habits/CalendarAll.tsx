import React, { useState } from 'react'
import { HabitHeader } from './HabitHeader'
import { Habit } from '../../utils/ModuleTypes'
import { Styled } from '../../styles/Calendar.styles'
import { ReactComponent as ChevronIcon } from '../../assets/icons/chevron.svg'
import { getCalendarDayInfo, getDayNextClass } from '../../utils/dateHelpers'
import { DayState } from '../../utils/ModuleTypes'
import {
  addWeeks,
  addDays,
  subWeeks,
  startOfWeek,
  endOfWeek,
  format,
  isSameDay
} from 'date-fns'

type CalendarProps = {
  habits: Habit[]
  handleDayClick: (
    habitId: string | number,
    day: Date,
    currState: DayState | null,
    dayId: null | string | number
  ) => void
}

const CalendarAll: React.FC<CalendarProps> = ({ habits, handleDayClick }) => {
  const [currentDate, setCurrentDate] = useState(new Date())

  const header = () => {
    const dateFormat = 'MMMM yyyy'

    return (
      <Styled.CalendarHeader>
        <Styled.CalendarHeader__Icon onClick={prevMonth}>
          <ChevronIcon />
        </Styled.CalendarHeader__Icon>
        <Styled.CalendarHeader__Title>
          {format(currentDate, dateFormat)}
        </Styled.CalendarHeader__Title>
        <Styled.CalendarHeader__Icon className="right" onClick={nextMonth}>
          <ChevronIcon />
        </Styled.CalendarHeader__Icon>
      </Styled.CalendarHeader>
    )
  }

  const handleClickDay = (
    e: any,
    habit: any,
    clondeDay: any,
    currState: any,
    dayId: any
  ) => {
    console.log(getDayNextClass(currState))
    const dayClass = getDayNextClass(currState)
    e.target.classList.add(dayClass)
    if (dayClass === 'blank') {
      e.target.classList.remove('done', 'not-done')
    }
    handleDayClick(habit, clondeDay, currState, dayId)
  }

  const daysOfWeek = () => {
    const dateFormat = 'E'
    const days = []
    let startDate = startOfWeek(currentDate)

    for (let i = 0; i < 7; i++) {
      days.push(
        <Styled.CalendarDOW__Day key={i}>
          {format(addDays(startDate, i), dateFormat)}
        </Styled.CalendarDOW__Day>
      )
    }

    return <Styled.CalendarDOW>{days}</Styled.CalendarDOW>
  }

  const cells = (habit: Habit) => {
    const rows = []
    let days = []
    let day = startOfWeek(currentDate)
    let formattedDate = ''

    while (day <= endOfWeek(currentDate)) {
      for (let i = 0; i < 7; i++) {
        formattedDate = format(day, 'd')
        const cloneDay = day
        let { dayClassName, currState, dayId } = getCalendarDayInfo(
          habit.days,
          cloneDay
        )

        days.push(
          <Styled.CalendarDays__Cell
            key={cloneDay.toString()}
            className={
              isSameDay(day, new Date())
                ? `today ${dayClassName}`
                : dayClassName
            }
            onClick={e =>
              handleClickDay(e, habit.id, cloneDay, currState, dayId)
            }
          >
            <Styled.CalendarDays__Cell__Inner>
              {formattedDate}
            </Styled.CalendarDays__Cell__Inner>
          </Styled.CalendarDays__Cell>
        )

        day = addDays(day, 1)
      }

      rows.push(
        <Styled.CalendarHabit key={habit.id}>
          <HabitHeader id={habit.id} title={habit.title} />
          <Styled.CalendarDays key={day.toString()}>
            {' '}
            {days}{' '}
          </Styled.CalendarDays>
        </Styled.CalendarHabit>
      )
      days = []
    }

    return rows
  }

  const nextMonth = () => {
    setCurrentDate(addWeeks(currentDate, 1))
  }

  const prevMonth = () => {
    setCurrentDate(subWeeks(currentDate, 1))
  }

  return (
    <Styled.CalendarContainer className="calendar">
      {header()}
      {daysOfWeek()}
      {habits.map(habit => cells(habit))}
    </Styled.CalendarContainer>
  )
}
export default CalendarAll
