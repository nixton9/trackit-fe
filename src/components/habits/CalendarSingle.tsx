import React, { useState } from 'react'
import { Habit } from '../../utils/ModuleTypes'
import { Styled } from '../../styles/Calendar.styles'
import { ReactComponent as ChevronIcon } from '../../assets/icons/chevron.svg'
import { getCalendarDayInfo, getDayNextClass } from '../../utils/dateHelpers'
import { DayState } from '../../utils/ModuleTypes'
import {
  addMonths,
  addDays,
  subMonths,
  startOfWeek,
  endOfWeek,
  startOfMonth,
  endOfMonth,
  format,
  isSameMonth,
  isSameDay
} from 'date-fns'

type CalendarProps = {
  habit: Habit
  handleDayClick: (
    habitId: string | number,
    day: Date,
    currState: DayState | null,
    dayId: null | string | number
  ) => void
}

const CalendarSingle: React.FC<CalendarProps> = ({ habit, handleDayClick }) => {
  const [currentDate, setCurrentDate] = useState(new Date())

  const header = () => {
    const dateFormat = 'MMMM yyyy'

    return (
      <Styled.CalendarHeader className="calendar-header">
        <Styled.CalendarHeader__Icon onClick={prevMonth}>
          <ChevronIcon />
        </Styled.CalendarHeader__Icon>
        <Styled.CalendarHeader__Title>
          {format(currentDate, dateFormat)}
        </Styled.CalendarHeader__Title>
        <Styled.CalendarHeader__Icon className="right" onClick={nextMonth}>
          <ChevronIcon />
        </Styled.CalendarHeader__Icon>
      </Styled.CalendarHeader>
    )
  }

  const handleClickDay = (
    e: any,
    habit: any,
    clondeDay: any,
    currState: any,
    dayId: any
  ) => {
    const dayClass = getDayNextClass(currState)
    e.target.classList.add(dayClass)
    if (dayClass === 'blank') {
      e.target.classList.remove('done', 'not-done')
    }
    handleDayClick(habit, clondeDay, currState, dayId)
  }

  const daysOfWeek = () => {
    const dateFormat = 'E'
    const days = []
    let startDate = startOfWeek(currentDate)

    for (let i = 0; i < 7; i++) {
      days.push(
        <Styled.CalendarDOW__Day key={i}>
          {format(addDays(startDate, i), dateFormat)}
        </Styled.CalendarDOW__Day>
      )
    }

    return <Styled.CalendarDOW>{days}</Styled.CalendarDOW>
  }

  const cells = () => {
    const monthStart = startOfMonth(currentDate)
    const rows = []
    let days = []
    let day = startOfWeek(monthStart)
    let formattedDate = ''

    while (day <= endOfWeek(endOfMonth(monthStart))) {
      for (let i = 0; i < 7; i++) {
        formattedDate = format(day, 'd')
        const cloneDay = day
        let isItSameMonth = isSameMonth(cloneDay, monthStart)
        let { dayClassName, currState, dayId } = getCalendarDayInfo(
          habit.days,
          cloneDay
        )

        days.push(
          <Styled.CalendarDays__Cell
            key={cloneDay.toString()}
            className={
              !isItSameMonth
                ? `disabled ${dayClassName}`
                : isSameDay(day, new Date())
                ? `today ${dayClassName}`
                : dayClassName
            }
            onClick={e => {
              if (isItSameMonth) {
                handleClickDay(e, habit.id, cloneDay, currState, dayId)
              }
            }}
          >
            <Styled.CalendarDays__Cell__Inner>
              {formattedDate}
            </Styled.CalendarDays__Cell__Inner>
          </Styled.CalendarDays__Cell>
        )

        day = addDays(day, 1)
      }

      rows.push(
        <Styled.CalendarDays key={day.toString()} className="calendar-single">
          {' '}
          {days}{' '}
        </Styled.CalendarDays>
      )
      days = []
    }

    return rows
  }

  const nextMonth = () => {
    setCurrentDate(addMonths(currentDate, 1))
  }

  const prevMonth = () => {
    setCurrentDate(subMonths(currentDate, 1))
  }

  return (
    <Styled.CalendarContainer className="calendar">
      {header()}
      {daysOfWeek()}
      {cells()}
    </Styled.CalendarContainer>
  )
}
export default CalendarSingle
