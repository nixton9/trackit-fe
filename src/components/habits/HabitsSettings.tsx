import React, { useState, useEffect } from 'react'
import Drawer from '../misc/Drawer'
import { SelectMenu } from '../misc/SelectMenu'
import { Styled } from '../../styles/Settings.styles'
import { useToggleElement } from '../../utils/useToggleElement'
import { ReactComponent as SettingsIcon } from '../../assets/icons/settings.svg'
import { useHistory } from 'react-router-dom'

const sortByOptions = [
  { val: 'date', label: 'Date' },
  { val: 'alphabetical', label: 'Alphabetical Order' }
]

const HabitsSettings: React.FC = () => {
  const [open, setOpen, overlayEl] = useToggleElement()
  const [sortBy, setSortBy] = useState('date')

  const history = useHistory()

  const handleSortByChange = (e: React.ChangeEvent<HTMLSelectElement>) =>
    setSortBy(e.target.value)

  useEffect(() => {
    history.push(`#`)

    window.onpopstate = (e: any) => {
      setOpen(false)
    }
  }, [history, setOpen])

  return (
    <>
      <SettingsIcon
        className="settings-icon"
        onClick={() => setOpen(true)}
        data-test-id="habits-settings-icon"
      />

      <Drawer
        title="Settings"
        open={open}
        setOpen={setOpen}
        overlayRef={overlayEl}
      >
        <Styled.SettingsBlock>
          <Styled.SettingsBlock__Label htmlFor="sort-by">
            Sort by
          </Styled.SettingsBlock__Label>
          <Styled.SettingsBlock__Input>
            <SelectMenu
              id="sort-by"
              value={sortBy}
              onChange={handleSortByChange}
              options={sortByOptions}
            />
          </Styled.SettingsBlock__Input>
        </Styled.SettingsBlock>
      </Drawer>
    </>
  )
}

export default HabitsSettings
