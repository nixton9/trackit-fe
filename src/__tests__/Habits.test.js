import React from 'react'
import { render } from '@testing-library/react'
import HabitsPage from '../components/habits/HabitsPage'
import CalendarAll from '../components/habits/CalendarAll'
import CalendarSingle from '../components/habits/CalendarSingle'
import HabitsSettings from '../components/habits/HabitsSettings'
import { habits } from '../assets/fakeData'
import { format } from 'date-fns'

describe('Habits Page', () => {
  it('matches snapshot', () => {
    const { container } = render(<HabitsPage />)

    expect(container).toMatchSnapshot()
  })

  it('has the title', () => {
    const { getByText } = render(<HabitsPage />)

    expect(getByText('Habits')).toBeInTheDocument()
  })
})

describe('CalendarAll', () => {
  it('matches snapshot', () => {
    const { container } = render(<CalendarAll habits={habits} />)

    expect(container).toMatchSnapshot()
  })

  it('shows current month and year', () => {
    const { getByText } = render(<CalendarAll habits={habits} />)
    const currDate = format(new Date(), 'MMMM yyyy')

    expect(getByText(currDate)).toBeInTheDocument()
  })

  it('has all the habits', () => {
    const { getByText } = render(<CalendarAll habits={habits} />)

    habits.forEach(habit => {
      expect(getByText(habit.title)).toBeInTheDocument()
    })
  })
})

describe('CalendarSingle', () => {
  it('matches snapshot', () => {
    const { container } = render(<CalendarSingle habit={habits[0]} />)

    expect(container).toMatchSnapshot()
  })

  it('shows current month and year', () => {
    const { getByText } = render(<CalendarSingle habit={habits[0]} />)
    const currDate = format(new Date(), 'MMMM yyyy')

    expect(getByText(currDate)).toBeInTheDocument()
  })

  it('shows all days', () => {
    const { getAllByText } = render(<CalendarSingle habit={habits[0]} />)

    for (let i = 1; i < 31; i++) {
      expect(getAllByText(i.toString())).toBeTruthy()
    }
  })
})

describe('HabitsSettings', () => {
  it('matches snapshot', () => {
    const { container } = render(<HabitsSettings />)

    expect(container).toMatchSnapshot()
  })

  it('has all the fields', () => {
    const { getByText } = render(<HabitsSettings />)

    expect(getByText('Sort by')).toBeInTheDocument()
  })
})
