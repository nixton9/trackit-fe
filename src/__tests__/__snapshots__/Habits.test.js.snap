// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarAll matches snapshot 1`] = `
<div>
  <div
    class="sc-AxjAm jfmAOU calendar"
  >
    <div
      class="sc-AxirZ geEfWa"
    >
      <span
        class="sc-AxhCb hJMRfN"
      >
        <svg>
          chevron.svg
        </svg>
      </span>
      <h2
        class="sc-AxiKw csrSaT"
      >
        September 2020
      </h2>
      <span
        class="sc-AxhCb hJMRfN right"
      >
        <svg>
          chevron.svg
        </svg>
      </span>
    </div>
    <div
      class="sc-AxhUy hfeXDI"
    >
      <h5
        class="sc-AxgMl iLoYic"
      >
        Sun
      </h5>
      <h5
        class="sc-AxgMl iLoYic"
      >
        Mon
      </h5>
      <h5
        class="sc-AxgMl iLoYic"
      >
        Tue
      </h5>
      <h5
        class="sc-AxgMl iLoYic"
      >
        Wed
      </h5>
      <h5
        class="sc-AxgMl iLoYic"
      >
        Thu
      </h5>
      <h5
        class="sc-AxgMl iLoYic"
      >
        Fri
      </h5>
      <h5
        class="sc-AxgMl iLoYic"
      >
        Sat
      </h5>
    </div>
    <div
      class="sc-AxheI iGtvIU"
    >
      <h4
        class="sc-Axmtr hMVYMT"
      >
        Fazer exerc√≠cio
      </h4>
      <div
        class="sc-AxmLO bELhKU"
      >
         
        <div
          class="sc-fzozJi ccINIa"
        >
          <span
            class="sc-fzoLsD gOMKUi"
          >
            27
          </span>
        </div>
        <div
          class="sc-fzozJi ccINIa today "
        >
          <span
            class="sc-fzoLsD gOMKUi"
          >
            28
          </span>
        </div>
        <div
          class="sc-fzozJi ccINIa"
        >
          <span
            class="sc-fzoLsD gOMKUi"
          >
            29
          </span>
        </div>
        <div
          class="sc-fzozJi ccINIa"
        >
          <span
            class="sc-fzoLsD gOMKUi"
          >
            30
          </span>
        </div>
        <div
          class="sc-fzozJi ccINIa not-done"
        >
          <span
            class="sc-fzoLsD gOMKUi"
          >
            1
          </span>
        </div>
        <div
          class="sc-fzozJi ccINIa"
        >
          <span
            class="sc-fzoLsD gOMKUi"
          >
            2
          </span>
        </div>
        <div
          class="sc-fzozJi ccINIa"
        >
          <span
            class="sc-fzoLsD gOMKUi"
          >
            3
          </span>
        </div>
         
      </div>
    </div>
    <div
      class="sc-AxheI iGtvIU"
    >
      <h4
        class="sc-Axmtr hMVYMT"
      >
        Fazer dieta
      </h4>
      <div
        class="sc-AxmLO bELhKU"
      >
         
        <div
          class="sc-fzozJi ccINIa"
        >
          <span
            class="sc-fzoLsD gOMKUi"
          >
            27
          </span>
        </div>
        <div
          class="sc-fzozJi ccINIa today "
        >
          <span
            class="sc-fzoLsD gOMKUi"
          >
            28
          </span>
        </div>
        <div
          class="sc-fzozJi ccINIa"
        >
          <span
            class="sc-fzoLsD gOMKUi"
          >
            29
          </span>
        </div>
        <div
          class="sc-fzozJi ccINIa"
        >
          <span
            class="sc-fzoLsD gOMKUi"
          >
            30
          </span>
        </div>
        <div
          class="sc-fzozJi ccINIa"
        >
          <span
            class="sc-fzoLsD gOMKUi"
          >
            1
          </span>
        </div>
        <div
          class="sc-fzozJi ccINIa"
        >
          <span
            class="sc-fzoLsD gOMKUi"
          >
            2
          </span>
        </div>
        <div
          class="sc-fzozJi ccINIa"
        >
          <span
            class="sc-fzoLsD gOMKUi"
          >
            3
          </span>
        </div>
         
      </div>
    </div>
  </div>
</div>
`;

exports[`CalendarSingle matches snapshot 1`] = `
<div>
  <div
    class="sc-AxjAm jfmAOU calendar"
  >
    <div
      class="sc-AxirZ geEfWa"
    >
      <span
        class="sc-AxhCb hJMRfN"
      >
        <svg>
          chevron.svg
        </svg>
      </span>
      <h2
        class="sc-AxiKw csrSaT"
      >
        September 2020
      </h2>
      <span
        class="sc-AxhCb hJMRfN right"
      >
        <svg>
          chevron.svg
        </svg>
      </span>
    </div>
    <div
      class="sc-AxhUy hfeXDI"
    >
      <h5
        class="sc-AxgMl iLoYic"
      >
        Sun
      </h5>
      <h5
        class="sc-AxgMl iLoYic"
      >
        Mon
      </h5>
      <h5
        class="sc-AxgMl iLoYic"
      >
        Tue
      </h5>
      <h5
        class="sc-AxgMl iLoYic"
      >
        Wed
      </h5>
      <h5
        class="sc-AxgMl iLoYic"
      >
        Thu
      </h5>
      <h5
        class="sc-AxgMl iLoYic"
      >
        Fri
      </h5>
      <h5
        class="sc-AxgMl iLoYic"
      >
        Sat
      </h5>
    </div>
    <div
      class="sc-AxmLO bELhKU"
    >
       
      <div
        class="sc-fzozJi ccINIa disabled "
      >
        <span
          class="sc-fzoLsD gOMKUi"
        >
          30
        </span>
      </div>
      <div
        class="sc-fzozJi ccINIa disabled "
      >
        <span
          class="sc-fzoLsD gOMKUi"
        >
          31
        </span>
      </div>
      <div
        class="sc-fzozJi ccINIa"
      >
        <span
          class="sc-fzoLsD gOMKUi"
        >
          1
        </span>
      </div>
      <div
        class="sc-fzozJi ccINIa"
      >
        <span
          class="sc-fzoLsD gOMKUi"
        >
          2
        </span>
      </div>
      <div
        class="sc-fzozJi ccINIa"
      >
        <span
          class="sc-fzoLsD gOMKUi"
        >
          3
        </span>
      </div>
      <div
        class="sc-fzozJi ccINIa"
      >
        <span
          class="sc-fzoLsD gOMKUi"
        >
          4
        </span>
      </div>
      <div
        class="sc-fzozJi ccINIa"
      >
        <span
          class="sc-fzoLsD gOMKUi"
        >
          5
        </span>
      </div>
       
    </div>
    <div
      class="sc-AxmLO bELhKU"
    >
       
      <div
        class="sc-fzozJi ccINIa"
      >
        <span
          class="sc-fzoLsD gOMKUi"
        >
          6
        </span>
      </div>
      <div
        class="sc-fzozJi ccINIa"
      >
        <span
          class="sc-fzoLsD gOMKUi"
        >
          7
        </span>
      </div>
      <div
        class="sc-fzozJi ccINIa"
      >
        <span
          class="sc-fzoLsD gOMKUi"
        >
          8
        </span>
      </div>
      <div
        class="sc-fzozJi ccINIa"
      >
        <span
          class="sc-fzoLsD gOMKUi"
        >
          9
        </span>
      </div>
      <div
        class="sc-fzozJi ccINIa"
      >
        <span
          class="sc-fzoLsD gOMKUi"
        >
          10
        </span>
      </div>
      <div
        class="sc-fzozJi ccINIa"
      >
        <span
          class="sc-fzoLsD gOMKUi"
        >
          11
        </span>
      </div>
      <div
        class="sc-fzozJi ccINIa"
      >
        <span
          class="sc-fzoLsD gOMKUi"
        >
          12
        </span>
      </div>
       
    </div>
    <div
      class="sc-AxmLO bELhKU"
    >
       
      <div
        class="sc-fzozJi ccINIa"
      >
        <span
          class="sc-fzoLsD gOMKUi"
        >
          13
        </span>
      </div>
      <div
        class="sc-fzozJi ccINIa"
      >
        <span
          class="sc-fzoLsD gOMKUi"
        >
          14
        </span>
      </div>
      <div
        class="sc-fzozJi ccINIa"
      >
        <span
          class="sc-fzoLsD gOMKUi"
        >
          15
        </span>
      </div>
      <div
        class="sc-fzozJi ccINIa"
      >
        <span
          class="sc-fzoLsD gOMKUi"
        >
          16
        </span>
      </div>
      <div
        class="sc-fzozJi ccINIa"
      >
        <span
          class="sc-fzoLsD gOMKUi"
        >
          17
        </span>
      </div>
      <div
        class="sc-fzozJi ccINIa"
      >
        <span
          class="sc-fzoLsD gOMKUi"
        >
          18
        </span>
      </div>
      <div
        class="sc-fzozJi ccINIa"
      >
        <span
          class="sc-fzoLsD gOMKUi"
        >
          19
        </span>
      </div>
       
    </div>
    <div
      class="sc-AxmLO bELhKU"
    >
       
      <div
        class="sc-fzozJi ccINIa"
      >
        <span
          class="sc-fzoLsD gOMKUi"
        >
          20
        </span>
      </div>
      <div
        class="sc-fzozJi ccINIa"
      >
        <span
          class="sc-fzoLsD gOMKUi"
        >
          21
        </span>
      </div>
      <div
        class="sc-fzozJi ccINIa"
      >
        <span
          class="sc-fzoLsD gOMKUi"
        >
          22
        </span>
      </div>
      <div
        class="sc-fzozJi ccINIa"
      >
        <span
          class="sc-fzoLsD gOMKUi"
        >
          23
        </span>
      </div>
      <div
        class="sc-fzozJi ccINIa"
      >
        <span
          class="sc-fzoLsD gOMKUi"
        >
          24
        </span>
      </div>
      <div
        class="sc-fzozJi ccINIa"
      >
        <span
          class="sc-fzoLsD gOMKUi"
        >
          25
        </span>
      </div>
      <div
        class="sc-fzozJi ccINIa"
      >
        <span
          class="sc-fzoLsD gOMKUi"
        >
          26
        </span>
      </div>
       
    </div>
    <div
      class="sc-AxmLO bELhKU"
    >
       
      <div
        class="sc-fzozJi ccINIa"
      >
        <span
          class="sc-fzoLsD gOMKUi"
        >
          27
        </span>
      </div>
      <div
        class="sc-fzozJi ccINIa today "
      >
        <span
          class="sc-fzoLsD gOMKUi"
        >
          28
        </span>
      </div>
      <div
        class="sc-fzozJi ccINIa"
      >
        <span
          class="sc-fzoLsD gOMKUi"
        >
          29
        </span>
      </div>
      <div
        class="sc-fzozJi ccINIa"
      >
        <span
          class="sc-fzoLsD gOMKUi"
        >
          30
        </span>
      </div>
      <div
        class="sc-fzozJi ccINIa disabled not-done"
      >
        <span
          class="sc-fzoLsD gOMKUi"
        >
          1
        </span>
      </div>
      <div
        class="sc-fzozJi ccINIa disabled "
      >
        <span
          class="sc-fzoLsD gOMKUi"
        >
          2
        </span>
      </div>
      <div
        class="sc-fzozJi ccINIa disabled "
      >
        <span
          class="sc-fzoLsD gOMKUi"
        >
          3
        </span>
      </div>
       
    </div>
  </div>
</div>
`;

exports[`Habits Page matches snapshot 1`] = `
<div>
  <div
    class="sc-fzpjYC izvAVX"
  >
    <h1
      class="sc-fznxsB iTQhax"
    >
      Habits
    </h1>
    <div
      class="sc-fznJRM cBJzUp"
    >
      <div
        class="sc-fznWqX dZCTFs"
      >
        <div
          class="sc-fzoiQi gKygiC"
        >
          <div
            class="MuiInputBase-root MuiInput-root"
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="habits-view"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
              id="habits-view"
              role="button"
              tabindex="0"
            >
              All
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              tabindex="-1"
              value="all"
            />
            <svg
              class="MuiSelect-icon"
            >
              chevron.svg
            </svg>
          </div>
        </div>
        <h3
          class="sc-fzqARJ lfDBcG smaller"
        >
          2
        </h3>
      </div>
      <div
        class="sc-fzqNqU hQTkrp"
      >
        <svg
          class="settings-icon"
        >
          settings.svg
        </svg>
        <div
          class="sc-fzpans bizqXm"
        >
          <h2
            class="sc-fznyAO eKnwhs"
          >
            Settings
          </h2>
          <div
            class="sc-fznKkj dFiY"
          >
            <div
              class="sc-fznZeY colRAS"
            >
              <label
                class="sc-fzokOt kgBGLn"
                for="sort-by"
              >
                Sort by
              </label>
              <div
                class="sc-fzqBZW ddlYPz"
              >
                <div
                  class="MuiInputBase-root MuiInput-root"
                >
                  <div
                    aria-haspopup="listbox"
                    aria-labelledby="sort-by"
                    class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                    id="sort-by"
                    role="button"
                    tabindex="0"
                  >
                    Date
                  </div>
                  <input
                    aria-hidden="true"
                    class="MuiSelect-nativeInput"
                    tabindex="-1"
                    value="date"
                  />
                  <svg
                    class="MuiSelect-icon"
                  >
                    chevron.svg
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-fzplWN chjtWg"
        />
      </div>
    </div>
  </div>
  <section
    class="sc-fzoyTs gUbeoJ"
  >
    <div
      class="sc-AxjAm jfmAOU calendar"
    >
      <div
        class="sc-AxirZ geEfWa"
      >
        <span
          class="sc-AxhCb hJMRfN"
        >
          <svg>
            chevron.svg
          </svg>
        </span>
        <h2
          class="sc-AxiKw csrSaT"
        >
          September 2020
        </h2>
        <span
          class="sc-AxhCb hJMRfN right"
        >
          <svg>
            chevron.svg
          </svg>
        </span>
      </div>
      <div
        class="sc-AxhUy hfeXDI"
      >
        <h5
          class="sc-AxgMl iLoYic"
        >
          Sun
        </h5>
        <h5
          class="sc-AxgMl iLoYic"
        >
          Mon
        </h5>
        <h5
          class="sc-AxgMl iLoYic"
        >
          Tue
        </h5>
        <h5
          class="sc-AxgMl iLoYic"
        >
          Wed
        </h5>
        <h5
          class="sc-AxgMl iLoYic"
        >
          Thu
        </h5>
        <h5
          class="sc-AxgMl iLoYic"
        >
          Fri
        </h5>
        <h5
          class="sc-AxgMl iLoYic"
        >
          Sat
        </h5>
      </div>
      <div
        class="sc-AxheI iGtvIU"
      >
        <h4
          class="sc-Axmtr hMVYMT"
        >
          Fazer exerc√≠cio
        </h4>
        <div
          class="sc-AxmLO bELhKU"
        >
           
          <div
            class="sc-fzozJi ccINIa"
          >
            <span
              class="sc-fzoLsD gOMKUi"
            >
              27
            </span>
          </div>
          <div
            class="sc-fzozJi ccINIa today "
          >
            <span
              class="sc-fzoLsD gOMKUi"
            >
              28
            </span>
          </div>
          <div
            class="sc-fzozJi ccINIa"
          >
            <span
              class="sc-fzoLsD gOMKUi"
            >
              29
            </span>
          </div>
          <div
            class="sc-fzozJi ccINIa"
          >
            <span
              class="sc-fzoLsD gOMKUi"
            >
              30
            </span>
          </div>
          <div
            class="sc-fzozJi ccINIa not-done"
          >
            <span
              class="sc-fzoLsD gOMKUi"
            >
              1
            </span>
          </div>
          <div
            class="sc-fzozJi ccINIa"
          >
            <span
              class="sc-fzoLsD gOMKUi"
            >
              2
            </span>
          </div>
          <div
            class="sc-fzozJi ccINIa"
          >
            <span
              class="sc-fzoLsD gOMKUi"
            >
              3
            </span>
          </div>
           
        </div>
      </div>
      <div
        class="sc-AxheI iGtvIU"
      >
        <h4
          class="sc-Axmtr hMVYMT"
        >
          Fazer dieta
        </h4>
        <div
          class="sc-AxmLO bELhKU"
        >
           
          <div
            class="sc-fzozJi ccINIa"
          >
            <span
              class="sc-fzoLsD gOMKUi"
            >
              27
            </span>
          </div>
          <div
            class="sc-fzozJi ccINIa today "
          >
            <span
              class="sc-fzoLsD gOMKUi"
            >
              28
            </span>
          </div>
          <div
            class="sc-fzozJi ccINIa"
          >
            <span
              class="sc-fzoLsD gOMKUi"
            >
              29
            </span>
          </div>
          <div
            class="sc-fzozJi ccINIa"
          >
            <span
              class="sc-fzoLsD gOMKUi"
            >
              30
            </span>
          </div>
          <div
            class="sc-fzozJi ccINIa"
          >
            <span
              class="sc-fzoLsD gOMKUi"
            >
              1
            </span>
          </div>
          <div
            class="sc-fzozJi ccINIa"
          >
            <span
              class="sc-fzoLsD gOMKUi"
            >
              2
            </span>
          </div>
          <div
            class="sc-fzozJi ccINIa"
          >
            <span
              class="sc-fzoLsD gOMKUi"
            >
              3
            </span>
          </div>
           
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`HabitsSettings matches snapshot 1`] = `
<div>
  <svg
    class="settings-icon"
  >
    settings.svg
  </svg>
  <div
    class="sc-fzpans bizqXm"
  >
    <h2
      class="sc-fznyAO eKnwhs"
    >
      Settings
    </h2>
    <div
      class="sc-fznKkj dFiY"
    >
      <div
        class="sc-fznZeY colRAS"
      >
        <label
          class="sc-fzokOt kgBGLn"
          for="sort-by"
        >
          Sort by
        </label>
        <div
          class="sc-fzqBZW ddlYPz"
        >
          <div
            class="MuiInputBase-root MuiInput-root"
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="sort-by"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
              id="sort-by"
              role="button"
              tabindex="0"
            >
              Date
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              tabindex="-1"
              value="date"
            />
            <svg
              class="MuiSelect-icon"
            >
              chevron.svg
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="sc-fzplWN chjtWg"
  />
</div>
`;
